### ISSUE TITLE
Linear Math from the bullet library will not build in Visual Studio 2017

### ISSUE THREAD

**Issue by Avynn** [hugoam/toy#6]
Maybe this is just me being dense but for whatever reason the linear math solution decides that some of the source files are missing even though when I go to check if they're there they are so I'm not sure why this is happening.  Either way I can't build the project.

here's the compiler output:
```
1>------ Build started: Project: LinearMath, Configuration: Debug Win32 ------
1>btAlignedAllocator.cpp
1>btConvexHull.cpp
1>btConvexHullComputer.cpp
1>c1xx : fatal error C1083: Cannot open source file: '..\..\..\3rdparty\bullet\src\LinearMath\btAlignedAllocator.cpp': No such file or directory
1>btGeometryUtil.cpp
1>c1xx : fatal error C1083: Cannot open source file: '..\..\..\3rdparty\bullet\src\LinearMath\btConvexHull.cpp': No such file or directory
1>c1xx : fatal error C1083: Cannot open source file: '..\..\..\3rdparty\bullet\src\LinearMath\btConvexHullComputer.cpp': No such file or directory
1>c1xx : fatal error C1083: Cannot open source file: '..\..\..\3rdparty\bullet\src\LinearMath\btGeometryUtil.cpp': No such file or directory
1>btPolarDecomposition.cpp
1>c1xx : fatal error C1083: Cannot open source file: '..\..\..\3rdparty\bullet\src\LinearMath\btPolarDecomposition.cpp': No such file or directory
1>btQuickprof.cpp
1>btSerializer.cpp
1>btSerializer64.cpp
1>btThreads.cpp
1>c1xx : fatal error C1083: Cannot open source file: '..\..\..\3rdparty\bullet\src\LinearMath\btThreads.cpp': No such file or directory
1>c1xx : fatal error C1083: Cannot open source file: '..\..\..\3rdparty\bullet\src\LinearMath\btQuickprof.cpp': No such file or directory
1>c1xx : fatal error C1083: Cannot open source file: '..\..\..\3rdparty\bullet\src\LinearMath\btSerializer.cpp': No such file or directory
1>c1xx : fatal error C1083: Cannot open source file: '..\..\..\3rdparty\bullet\src\LinearMath\btSerializer64.cpp': No such file or directory
1>btVector3.cpp
1>c1xx : fatal error C1083: Cannot open source file: '..\..\..\3rdparty\bullet\src\LinearMath\btVector3.cpp': No such file or directory
1>Done building project "LinearMath.vcxproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
```

**Comment by hugoam** [comment]
Yep it's just a problem when cloning. You have to clone with `--recursive`.
If you didn't you have to run `git submodule update --init`, first in `toy`, and then in `toy/mud`

**Comment by Avynn** [comment]
Yeah I'll build those individually and get back to you in a sec

**Comment by Avynn** [comment]
The build output for mud was way too long to copy and paste here so instead I took a screen cap of the output maybe that'll help more,  I can copy and paste the output as well if you want.
![screenshot 3](https://user-images.githubusercontent.com/22106487/43616691-a59dcef0-9672-11e8-8989-62b553cb06da.png)

**Comment by hugoam** [comment]
So it can't basically open any file in the whole codebase. Can you even open one of the files in the projects from visual studio ?
The only thing to check is that the relative path indicated here indeed leads to the file, from the project directory (it should). Apart from an access rights issue, I don't know what could be causing this. Did you try running Visual Studio as administrator ?

**Comment by Avynn** [comment]
I checked the properties of the folder containing the source and everything looks fine all the access rights should work.  Regardless I ran VS as admin and got the same compilation result.  I can open view and modify the source files as well within the solution but I do get an interesting active error the code is E1696 and the message is :
```
Error (active)	E1696	cannot open source file "E:\TOYIOENGINE\TOY\3RDPARTY\BULLET\SRC\BULLETCOLLISION\BROADPHASECOLLISION\BTAXISSWEEP3.CPP"
```
the path to the file in the error message is always to whatever file I happen to have open.
